web/src/buddy_data.js

```diff
@@ -32,10 +32,18 @@ let is_searching_users = false;
 
 // TODO: This should ideally be in buddy_list.js since it uses jquery,
 // but can't be easily placed there due to import cycles.
-function update_empty_list_placeholders() {
-    const empty_list_message = is_searching_users
-        ? $t({defaultMessage: "No matching users."})
-        : $t({defaultMessage: "None."});
+export function update_empty_list_placeholders(
```

Comment:
Darn, this commit is actually not quite right because the call to `update_empty_list_placeholders` in this file doesn't have the arguments to pass to it. So this will break if "No active users" is shown and the user searches and then stops searching.

I can't think of a cleaner way to do this than to have variables in this file that keep track of if there are inactive users, or I guess a function that calculates those variables from scratch (but from a separate module that's imported from both here and buddy list, to avoid circular imports)? Thoughts?

Reply:
Aha I figured out a solution, which makes everything nicer. I realized that the `populate` function is called every time the search string changes, to repopulate how the users are displayed, so I can just do the empty list stuff only from the populate function. I'll make some changes and push that.